straxen/url_config.py

```diff
@@ -8,11 +8,11 @@
 from urllib.parse import urlparse, parse_qs
 from ast import literal_eval
 from functools import lru_cache
-
 from strax.config import OMITTED
 
 export, __all__ = strax.exporter()
 
+_CACHES = {}
```

Comment:
is there a way of preventing this thing from becoming enormous, we don't do garbage collection right? I might be missing some key features, but it seems to me that only the values that are stored in _CACHES are checked for not becoming too long, whereas I would have expected we would also have to check that _CACHES itself doesn't become too large

Reply:
well the size of _CACHES is limited by the number of plugin classes, each plugin class will have one instance of the URLConfig for each of its configs. So while its trivial to swtich _CACHES to a CachingDict instead of a regular dict, I didn't think it would make any difference, how many cached configs can you already have?

Reply:
good question, but I honestly don't know how many we (are going) to have. 

Maybe we can add a warning to where we append new things to _CACHES to check that `sys.sizeof` `_CACHE` and warns that someone might want to free some memory by doing a `_CACHE={}`  or something if the size is bigger than 1GB.

Maybe this warning will never come, but memory leaks are such a messy business to fix especially if they show up in a year from now when we have some crazy map we need to open or something.

Reply:
True, maybe i should just replace the LRU cache with a TTL cache?

Reply:
Sounds good, although I don't oppose this LRU implementation either. It's a first and useful version and we might have learn to learn what works best for us.

I just want to prevent making a mistake like I did here: https://github.com/AxFoundation/strax/pull/561 that pops up silently in ways I did not image people using straxen (which was a clearly a limitation of my imagination, not of the people I should add)

Reply:
OK so for now i added  `config_cache_size_mb` and  `clear_config_caches` functions. If we want to switch to a TTL cache in the future that would be pretty easy.